rethinkdb/datadog_checks/rethinkdb/document_db/query.py

```diff
@@ -0,0 +1,118 @@
+
+        path = spec['path']
+        name = spec.get('name', path)
+        value = lookup_dotted(document, path=path)
+
+        if 'modifier' in spec and spec['modifier'] is not None:
+            value = self._modify(value, modifier=spec['modifier'], logger=logger)
+
+        if not isinstance(value, (int, float)):  # pragma: no cover
+            raise RuntimeError('Expected float or int, got {!r} of type {}', value, type(value))
+
+        name = dotted_join((self.prefix, name))
+
+        return {'type': spec['type'], 'name': name, 'value': value, 'tags': tags}
+
+    def _make_metrics_from_enumeration(self, document, enumeration, tags, logger):
+        # type: (Any, Enumeration, List[str], CheckLoggingAdapter) -> Iterator[Metric]
+        logger.trace('make_metrics_from_enumeration enumeration=%r', enumeration)
+
+        values = lookup_dotted(document, path=enumeration['path'])  # type: Sequence
+
+        for index, value in enumerate(values):
+            item_tags = tags + ['{}:{}'.format(enumeration['index_tag'], index)]
+
+            for spec in enumeration['metrics']:
+                spec = {
+                    'type': spec['type'],
+                    'name': dotted_join((enumeration['path'], spec['path'])),
+                    'path': spec['path'],
+                    'modifier': spec.get('modifier'),
+                }
+                yield self._make_metric_from_spec(value, spec, tags=item_tags, logger=logger)
+
+    def _make_metrics_from_group(self, document, group, tags, logger):
+        # type: (Any, Group, List[str], CheckLoggingAdapter) -> Iterator[Metric]
+        logger.trace('make_metrics_from_group group=%r', group)
+
+        mapping = lookup_dotted(document, path=group['path'])  # type: Mapping
+
+        for key in mapping:
+            item_tags = tags + ['{}:{}'.format(group['key_tag'], key)]
+            spec = {
+                'type': group['value_metric_type'],
+                'name': group['path'],
+                'path': key,
+            }  # type: MetricSpec
+            yield self._make_metric_from_spec(mapping, spec, tags=item_tags, logger=logger)
+
+    def _modify(self, value, modifier, logger):
+        # type: (Any, Modifier, CheckLoggingAdapter) -> float
+        logger.trace('modify value=%r modifier=%r', value, modifier)
+
+        if modifier == 'total':
+            return len(value)
+
+        if modifier == 'ok_warning':
+            return AgentCheck.OK if value else AgentCheck.WARNING
+
+        if modifier == 'time_elapsed':
+            return to_time_elapsed(value)
+
+        raise RuntimeError('Unknown modifier: {!r}'.format(modifier))  # pragma: no cover
+
+    def run(self, *args, **kwargs):
+        # type: (*Any, **Any) -> Iterator[Metric]
+        logger = kwargs.pop('logger')  # type: CheckLoggingAdapter
+
+        logger.debug('query_%s', self.name)
+
+        for document, tags in self.source(*args, **kwargs):
```

Comment:
```suggestion
        for document, tags in self.source(conn):
```

Could be explicit + seem the kwargs is never used.

Related to comment above.

Reply:
See my answer here https://github.com/DataDog/integrations-core/pull/5715#discussion_r397839898